src/utils/formatBalance.ts

```diff
@@ -62,13 +63,53 @@ export const formatFixedNumber = (number: ethers.FixedNumber, displayDecimals =
   return formatBigNumber(ethers.BigNumber.from(leftSide), displayDecimals, decimals)
 }
 
-export const formatLocalisedCompactNumber = (number: number): string => {
+export const formatLocalisedCompactNumber = (
+  number: number,
+  t: (key: string, data?: ContextData) => string,
+): string => {
   const codeFromStorage = getLanguageCodeFromLS()
+  const hasNotFormatSupport = typeof Intl === 'undefined' || !('NumberFormat' in Intl) || !supportsES2020()
+
+  if (hasNotFormatSupport) {
+    return nFormatter(number, t)
+  }
+
   return new Intl.NumberFormat(codeFromStorage, {
     notation: 'compact',
     compactDisplay: 'long',
     maximumSignificantDigits: 2,
   }).format(number)
 }
 
-export default formatLocalisedCompactNumber
+const nFormatter = (num, t) => {
+  if (num >= 1000000000) {
+    return `${(num / 1000000000).toFixed(1).replace(/\.0$/, '')} ${t('Billion').toLocaleLowerCase()}`
+  }
+  if (num >= 1000000) {
+    return `${(num / 1000000).toFixed(1).replace(/\.0$/, '')} ${t('Million').toLocaleLowerCase()}`
+  }
+  if (num >= 1000) {
+    return `${(num / 1000).toFixed(1).replace(/\.0$/, '')} ${t('Thousand').toLocaleLowerCase()}`
+  }
+  return num
+}
```

Comment:
True, thats why i keep the original numberformat for new versions of safari. I thought it is better to show some kind of formatted text that fits many languages instead of number. Or we can have the polyfill approach with the issue that @ChefHutch showed in video.

Reply:
I'm not sure this is any better Memoyil?

We're going from a correctly localised string - albeit the full currency number, to an incorrectly localised text number.

Personally I think the fallback behaviour (i.e. showing the full number) is preferable to an incorrectly localised bit of text, that brings with it this logic weight.

If it isn't possible to make the polyfilling performant - let's leave the code as-is.
If there are display issues with showing the full number - let's fix them.

Reply:
Done 👍 Can you review it again?